### ISSUE TITLE
Commands don't execute

### ISSUE THREAD

**Issue by JohnWilliston** [doctorfree/cheatsheet.nvim#7]
I feel like I must be doing something wrong. I use Telescope and was easily able to make the plugin work beautifully via the following bit of configuration:

```
return {
    "doctorfree/cheatsheet.nvim",
    event = "VeryLazy",
    dependencies = {
        { "nvim-telescope/telescope.nvim" },
        { "nvim-lua/popup.nvim" },
        { "nvim-lua/plenary.nvim" },
    },
    config = function()
        local ctactions = require("cheatsheet.telescope.actions")
        require("cheatsheet").setup({
            bundled_cheatsheets = true,
            bundled_plugin_cheatsheets = true,
            include_only_installed_plugins = true,
            telescope_mappings = {
                ["<CR>"] = ctactions.select_or_fill_commandline,
                ["<A-CR>"] = ctactions.select_or_execute,
                ["<C-Y>"] = ctactions.copy_cheat_value,
                ["<C-E>"] = ctactions.edit_user_cheatsheet,
            },
        })
    end,
}
```

When I hit `<leader>?`, I get the nice telescope list and can search everything just fine. But then pressing enter or alt enter on a selected command does absolutely nothing. Is there something I'm missing?

**Comment by JohnWilliston** [comment]
Ok, never mind, I've solved my own problem. Some debugging showed that it was a two-fold problem:

1. The default configuration (which I copied) binds `ctactions.select_or_fill_commandline` to the enter key, and that explicitly and deliberately _doesn't_ execute the command. I'm honestly not sure what it does as I can't get it to do anything with or without the command line. Not sure why.
2. The default configuration supplies an alternate method of invoking via `<A-CR>`, but apparently the Windows Terminal program sucks up that keystroke and doesn't let it pass. But if I rebind that to something like `<F2>` then my command will be executed as anticipated.

I have to say I really don't understand the readme as written. Maybe it's obvious to others. But once I rebound the keys as described, I can get the behavior I was expecting save for nothing happening with the command line and the `copy_cheat_value` not seeming to work either. Not sure what to think at this point.

**Comment by chip0tle** [comment]
@JohnWilliston hey John - i'm not able to get the copy_cheat_value keymap to actually work either. i managed to get it to trigger one single time, but cannot for the life of me replicate it. did you ever come up with any fixes?

**Comment by JohnWilliston** [comment]
> [@JohnWilliston](https://github.com/JohnWilliston) hey John - i'm not able to get the copy_cheat_value keymap to actually work either. i managed to get it to trigger one single time, but cannot for the life of me replicate it. did you ever come up with any fixes?

Yeah, let me share a few things you might find helpful after which I'll paste my current config that will (hopefully) work a tad better for you than what you have now. First, I suspect the copy function is working just fine as the author intended, but I'm not sure his intent kept pace with Neovim's development. I say that because I think you'll find the setup you have now copies the text alright _but only to the " register_. I integrate Neovim with the system clipboard and prefer the copy function to include the * register as well for that reason, so I made my own customization to the plugin.

Second, I'm honestly not sure what the author's intent was for the `select_or_fill_commandline` function because all the code does for that path is assemble some fancy chunks of text with highlight-group names and then send it to the `vim.api.nvim_echo()` function. I'm a bit beyond my depth here, but I find under normal circumstances when I call that function from the command line it prints what you send it to the message line (or whatever it's properly called) at the bottom of the editor. But for some reason, it doesn't show me _anything_ when called from the plugin's code. My own investigation showed that the author was also passing a `false` parameter for whether the resulting text should be included in the systems message history (viewed by using the `:messages` command). I changed that to `true` in my version of the plugin for both cases.

So short version of the story: if you think you might want the changes I've described, you can check out [my fork of the plugin here](https://github.com/JohnWilliston/cheatsheet.nvim). I've opened a pull request but doubt the author will accept it. If you know more about how the `nvim_echo()` function works (I find differences between stock Neovim and Neovide for example), please share what you know. I'm also open to hearing how it _should_ work instead, as I really like the cheat sheet plugin and could certainly add more to my fork of it. And for your (and others) convenience, my plugin spec for my own version of it is as follows:

```language-lua
return {
    "JohnWilliston/cheatsheet.nvim",
    event = "VeryLazy",
    dependencies = {
        "nvim-telescope/telescope.nvim",
        "nvim-lua/popup.nvim",
        "nvim-lua/plenary.nvim",
    },
    keys = {
        { "<leader>??", "<cmd>Cheatsheet<cr>", desc = "Cheat sheet" }
    },
    opts = function ()
        require("telescope").load_extension("cheatsheet")
        local ctactions = require("cheatsheet.telescope.actions")
        
        return {
            bundled_cheatsheets = true,
            bundled_plugin_cheatsheets = true,
            include_only_installed_plugins = true,
            telescope_mappings = {
                ["<CR>"] = ctactions.select_or_execute,
                -- Sends text description of cheat to nvim_echo() method.
                -- If I use my altered version of the plugin, it shows up in
                -- the message history. Not sure of the author's intent.
                ["<C-k>"] = ctactions.select_or_fill_commandline,
                ["<C-y>"] = ctactions.copy_cheat_value,
                -- Not needed as it's the default, but let's be explicit.
                ["<C-e>"] = ctactions.edit_user_cheatsheet,
            },
        }
    end,
}
```
